ui/components/ui/box/box.js

```diff
@@ -104,54 +88,113 @@ export const ValidBorderColors = [
-  const arr = Array.isArray(main) ? main : [];
-  const singleDigit = Array.isArray(main) ? undefined : main;
-  if (Array.isArray(main) && ![2, 3, 4].includes(main.length)) {
-    throw new Error(
-      `Expected prop ${type} to have length between 2 and 4, received ${main.length}`,
-    );
-  }
+function isValidString(type, value) {
+  return typeof type === 'string' && typeof value === 'string';
+}
 
-  const isHorizontalAndVertical = arr.length === 2;
-  const isTopHorizontalAndBottom = arr.length === 3;
-  const isAllFour = arr.length === 4;
-  const hasAtLeastTwo = arr.length >= 2;
-  const hasAtLeastThree = arr.length >= 3;
+/**
+ * Generates classnames for different utility styles from utility props such as margin, padding, display, etc.
+ * Accepts numbers and strings for static values and arrays for responsive values.
+ * Maps responsive props to mobile first breakpoints
+ * Returns an object for 'classnames' package to generate the utility classNames
+ *
+ * @param {string} baseClass - The root or base class name
+ * @param {string} type - The style declaration type "margin", "margin-top", "padding", "display" etc
+ * @param {array || number || string} value - prop value being passed in array props are responsive props
+ * @param {*} validatorFn - The validation function for each value type
+ * @returns
+ */
 
-  return {
-    [`${baseClass}--${type}-${singleDigit}`]: isValidValue(type, singleDigit),
-    [`${baseClass}--${type}-top-${top}`]: isValidValue(type, top),
-    [`${baseClass}--${type}-right-${right}`]: isValidValue(type, right),
-    [`${baseClass}--${type}-bottom-${bottom}`]: isValidValue(type, bottom),
-    [`${baseClass}--${type}-left-${left}`]: isValidValue(type, left),
-    // As long as an array of length >= 2 has been provided, the first number
-    // will always be for the top value.
-    [`${baseClass}--${type}-top-${arr?.[0]}`]: hasAtLeastTwo,
-    // As long as an array of length >= 2 has been provided, the second number
-    // will always be for the right value.
-    [`${baseClass}--${type}-right-${arr?.[1]}`]: hasAtLeastTwo,
-    // If an array has 2 values, the first number is the bottom value. If
-    // instead if has 3 or more values, the third number will be the bottom.
-    [`${baseClass}--${type}-bottom-${arr?.[2]}`]: hasAtLeastThree,
-    [`${baseClass}--${type}-bottom-${arr?.[0]}`]: isHorizontalAndVertical,
-    // If an array has 2 or 3 values, the second number will be the left value
-    [`${baseClass}--${type}-left-${arr?.[1]}`]:
-      isHorizontalAndVertical || isTopHorizontalAndBottom,
-    // If an array has 4 values, the fourth number is the left value
-    [`${baseClass}--${type}-left-${arr?.[3]}`]: isAllFour,
-  };
+function generateClassNames(baseClass, type, value, validatorFn) {
+  // if value does not exist return null
+  if (!value) {
+    return null;
+  }
+  // initiate classObject to be returned to classnames package
+  let classesObject = {};
```

Comment:
instead of using `let` and then creating a new object every time by spreading the old object into it, why not just use a constant and declare the properties on that object if needed? In terms of space complexity this will reduce the amount of memory footprint the method requires (until the function is complete, every new object will be kept in memory until GC can determine that there is no longer any references pointing to the old object)

Reply:
Updated: https://github.com/MetaMask/metamask-extension/pull/15106/commits/d1e4233847681f2130fffd4bd863237a1da608cc